File: engine_py/tests/test_job_type_metadata.py
def test_get_job_type_metadata(make_world):
    world = make_world()
    job_types = world.get_job_types()
    assert isinstance(job_types, list)
    assert "DigTunnel" in job_types

    meta = world.get_job_type_metadata("DigTunnel")
    assert isinstance(meta, dict)
    assert meta["name"] == "DigTunnel"
    assert "effects" in meta
    assert isinstance(meta["effects"], list)
    assert "requirements" in meta
    assert meta["requirements"] == ["Tool:Pickaxe"]
    assert meta["duration"] == 5
    assert meta["effects"][0]["action"] == "ModifyTerrain"
    assert meta["effects"][0]["from"] == "rock"
    assert meta["effects"][0]["to"] == "tunnel"

File: engine/scripts/lua/tests/test_job_type_metadata.lua
local assert = require("assert")

local function test_get_job_type_metadata()
	local job_types = get_job_types()
	assert.is_table(job_types)
	assert.contains("DigTunnel", job_types)

	local meta = get_job_type_metadata("DigTunnel")
	assert.is_table(meta)
	assert.equals(meta.name, "DigTunnel")
	assert.is_table(meta.effects)
	assert.is_table(meta.requirements)
	assert.equals(meta.requirements[1], "Tool:Pickaxe")
	assert.equals(meta.duration, 5)
	assert.equals(meta.effects[1].action, "ModifyTerrain")
	assert.equals(meta.effects[1].from, "rock")
	assert.equals(meta.effects[1].to, "tunnel")
end

return {
	test_get_job_type_metadata = test_get_job_type_metadata,
}
